### PR TITLE
fix: standardise filename before testing format support

### LINKED ISSUE #12194: [YAFW] Capitalized extensions are not seen as correct file type
### Extension

https://www.raycast.com/pablopunk/yafw

### Description

I took a screen recording on my iPhone which I air dropped to my Macbook. The file had the extension `.MP4` (all caps). The extension said that it was not a valid video file type despite `.mp4` being in the list of suggested types. 

In Finder I changed `.MP4` to `.mp4` and re-ran the extension. It worked as expected at this point. 

### Steps To Reproduce

1. Create (or rename) an mp4 with the extension `.MP4` (all caps).
2. Select the file in Finder and trigger the extension in Raycast.
3. The extension will throw an error saying that the file type is not supported.
4. Change the file extension to `.mp4` (lowercase).
5. Trigger the extension again. This time it will work and compress the video as expected. 

### Current Behaviour

Does not recognize `.MP4` files as valid.

### Expected Behaviour

I would expect that the extension would take into account whether the file extension is lower or upper case.

### PR DESCRIPTION
**Pull request by mlz11**
## Description

This PR is a proposal to fix #12194.

## Screencast

Now attempting to compress .MP4, .MOV etc. files should work properly.

https://github.com/raycast/extensions/assets/94069699/a9c312d1-017f-4533-a998-5dc5d514decb

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/yafw/src/utils.tsx [hunk 1]
```diff
@@ -109,3 +109,7 @@ export function capitalizeSnakeCase(text: string): string {
     .map((part) => part.charAt(0).toUpperCase() + part.slice(1))
     .join(" ");
 }
+
+export function isFileFormatSupported(filename: string): boolean {
+  return VIDEO_FORMATS.some((format) => filename.toLowerCase().endsWith(format));
+}
```
Comment (pablopunk) [inline extensions/yafw/src/utils.tsx:115]:
```suggestion
export function isFileFormatSupported(_filename: string): boolean {
  const filename = _filename.toLowerCase()
  return VIDEO_FORMATS.some((format) => filename.endsWith(format));
}
```

so we don't repeat "toLowerCase()" once per extension

**Review by pablopunk — APPROVED** [review]
Thanks for the fix! 🥳

File: extensions/yafw/src/utils.tsx (hunk 1 above)
Comment (mlz11) [reply extensions/yafw/src/utils.tsx]: good catch thanks :) 

I applied your suggestion

**Comment by tonypinkevych** [comment]
@mlz11 hi, thanks for the PR.

Can you please add your function to this line too?
https://github.com/raycast/extensions/blob/3a03fcd82e0205e0cfb4688793a93cf5675d896c/extensions/yafw/src/objects/selected-finder.files.ts#L23

This is related to my recent updates. And since @pablopunk and I created similar extensions, I had to merge all my code with `yafw` extension, which resulted in temporary duplicates. I will create new PR with some refactor soon.

**Comment by mlz11** [comment]
> @mlz11 hi, thanks for the PR.
> 
> Can you please add your function to this line too?
> 
> https://github.com/raycast/extensions/blob/3a03fcd82e0205e0cfb4688793a93cf5675d896c/extensions/yafw/src/objects/selected-finder.files.ts#L23
> 
> This is related to my recent updates. And since @pablopunk and I created similar extensions, I had to merge all my code with `yafw` extension, which resulted in temporary duplicates. I will create new PR with some refactor soon.

Hello @tonypinkevych 👋🏻 
Thanks for your comment, it's done ✅ Does it look good to you ?

It's my first contribution to Raycast Extensions. I still have trouble understanding how to keep a clean git history and am a bit ashamed at how this PR looks 😅 Do you have any advice ?

**Comment by tonypinkevych** [comment]
@mlz11 yes, thanks a lot! 

Sorry to bother you, but could you also add `.gif` to this array https://github.com/raycast/extensions/blob/11747cadee7366bc688f808c1a83ce861f2b66f8/extensions/yafw/src/constants.tsx#L12? 😅
My part supports gif, but the core part does not. I think it will be ok to add gif to both parts for now.

It's the last comment from my side.

**Comment by tonypinkevych** [comment]
> It's my first contribution to Raycast Extensions. I still have trouble understanding how to keep a clean git history and am a bit ashamed at how this PR looks 😅 Do you have any advice ?

This is exactly how raycast has the system set up. But all commits will be squashed during the merge. So there's no problem. Just ignore it.

**Comment by mlz11** [comment]
> > Thanks for your comment, it's done ✅ Does it look good to you ?
> 
> @mlz11 yes, thanks a lot!
> 
> Sorry to bother you, but could you also add `.gif` to this array? 😅
> 
> https://github.com/raycast/extensions/blob/11747cadee7366bc688f808c1a83ce861f2b66f8/extensions/yafw/src/constants.tsx#L12
> 
> 
> My part supports gif, but the core part does not. I think it will be ok to add gif to both parts for now.
> It's the last comment from my side.

Done @tonypinkevych ✅
It feels a bit weird when compressing a `.gif` file to obtain a `.mp4` file but I think it's ok for now. @pablopunk can you confirm you're OK with it ?

I feel like it's worth spending a little bit of time understanding how to compress `.gif → .gif` using `ffmpeg`. I'll try and do that soon :)

**Comment by pernielsentikaer** [comment]
Is this PR ready to be merged or are we still looking into fixes

**Comment by mlz11** [comment]
> Is this PR ready to be merged or are we still looking into fixes

Hello @pernielsentikaer 👋🏻
I prefer to wait for a last approval from @pablopunk on 92b5929ba6b9534059fa57670c4444aa25faabd4 😊

File: extensions/yafw/src/constants.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ export const COMPRESSION_OPTIONS: Record<CompressionOptionKey, CompressionOption
   very_small_file: { crf: 42, bitrate: "500k", bufsize: "1000k" },
 };
 export const DEFAULT_COMPRESSION: CompressionOptionKey = "balanced";
-export const VIDEO_FORMATS = ["mp4", "mov", "avi", "mkv", "webm"];
+export const VIDEO_FORMATS = ["mp4", "mov", "avi", "mkv", "webm", "gif"];
```
[the remark this answers is not in the document]
Comment (mlz11) [reply extensions/yafw/src/constants.tsx:12]: Hello @pablopunk yes it does. I have tested it (see attachment below).

https://github.com/raycast/extensions/assets/94069699/993fc17c-17b3-4438-8692-a67796f18b8a

Nonetheless, as I was saying in the comment below I just wanted you to confirm that you're OK with the following behavior :

> It feels a bit weird when compressing a .gif file to obtain a .mp4 file but I think it's ok for now. @pablopunk can you confirm you're OK with it ?
> 
> I feel like it's worth spending a little bit of time understanding how to compress .gif → .gif using ffmpeg. I'll try and do that soon :)

**Comment by pernielsentikaer** [comment]
Is there any outstanding issues in this PR before we can merge it?

**Review by pablopunk — APPROVED** [review]

**Comment by pablopunk** [comment]
@pernielsentikaer let's merge it! 🎉

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**